### PR TITLE
add iterator for msgs access of mulitpart_t

### PR DESCRIPTION
**Pull request by laplaceyang**
add iterator for msgs access of mulitpart_t

### CODE REVIEW

File: zmq_addon.hpp [hunk 1]
```diff
@@ -87,6 +93,47 @@ class multipart_t
...
     }
 
+    iterator begin()
+    {
+        return m_parts.begin();
+    }
+
+    const_iterator begin() const
+    {
+        return m_parts.begin();
```
Comment (pavel-pimenov) [inline zmq_addon.hpp]: .cbegin() ?

Comment (laplaceyang) [inline zmq_addon.hpp]: you are right

**Comment by bluca** [comment]
Could you please squash the last commit into the one that introduced the issue?

**Comment by laplaceyang** [comment]
I will squash them. I'm sorry  I am new here.

**Comment by bluca** [comment]
No problem! Thanks for your contributions

**Comment by bluca** [comment]
@laplaceyang there's a merge conflict so I can't merge it, could you please rebase?

In case you are new to the github workflow:

```
git remote add upstream https://github.com/zeromq/cppzmq.git
git fetch upstream
git rebase --onto upstream/master HEAD^
git push -f
```

**Comment by laplaceyang** [comment]
@bluca I am trying to do this..

**Comment by laplaceyang** [comment]
@bluca It seems all thing has been done , thank you so much for your guide

**Comment by bluca** [comment]
No problem, thanks for your contributions!

**Pull request merged**